packages/api/src/API.ts

```diff
@@ -325,11 +325,11 @@ export default class APIClass {
      * @param {GraphQLOptions} GraphQL Options
      * @returns {Promise<GraphQLResult> | Observable<object>}
      */
-    graphql({ query, variables = {} }: GraphQLOptions) {
+    graphql({ query: paramQuery, variables = {} }: GraphQLOptions) {
 
-        const doc = parse(query);
+        const query = typeof paramQuery === 'string' ? parse(paramQuery) : paramQuery;
```

Comment:
I would validate if paramQuery is a valid AST

```suggestion
        const query = typeof paramQuery === 'string' ? parse(paramQuery) : parse(print(paramQuery));
```